Automattic/jetpack
# AI Client: introduce audio duration display component

This PR introduces the audio duration display component for the Create with voice block.
Getting the audio duration when recording the audio from the browser gets tricky. I've implemented a helper function to deal with this, based on this [StackOverflow thread](https://stackoverflow.com/questions/21522036/html-audio-tag-duration-always-infinity). We can keep researching other ways to compute the audio duration time in follow-ups.

So, in this PR:
* Add the getDuration() helper function
* Add formatTime() helper function
* Introduce the `<AudioDurationDisplay />` component


Fixes https://github.com/Automattic/jetpack/issues/32794

## Proposed changes:

* AI Client: introduce audio duration display component

### Other information:

## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Test with StoryBook

```
cd projects/js-packages/storybook/
```

```
pnpm run storybook:dev
```

* Go to the component story: `http://localhost:50240/?path=/story/js-packages-ai-client-usemediarecording--default`
* Start recording audio
* Confirm the component shows the recording time

https://github.com/Automattic/jetpack/assets/77539/c6eb549d-63f3-48ae-b1c3-d14cadae1208
